Dalamud/Game/Gui/Nameplates/NameplateGui.cs

```diff
@@ -92,72 +91,58 @@ public void DisposeService()
-                IconID = (StatusIcons)iconId,
+                IconID = (NameplateStatusIcons)iconId,
             };
 
+            // Add known nameplate nodes
+            nameplateInfo.Nodes.Add(new(titlePtr, NameplateNodeName.Title));
+            nameplateInfo.Nodes.Add(new(namePtr, NameplateNodeName.Name));
+            nameplateInfo.Nodes.Add(new(freeCompanyPtr, NameplateNodeName.FreeCompany));
+            nameplateInfo.Nodes.Add(new(prefixPtr, NameplateNodeName.Prefix));
+
+            // Create new nameplate object
+            var namePlateObj = new NameplateObject(playerNameplateObjectPtr, nameplateInfo);
+
             // Invoke event
-            var eventArgs = new NameplateUpdateEventArgs(nameplateInfo, namePlateObj);
-            this.OnNameplateUpdate.Invoke(eventArgs);
+            this.OnNameplateUpdate.Invoke(namePlateObj);
 
-            if (eventArgs.HasChanged)
+            // Get new states
+            isTitleAboveName = nameplateInfo.IsTitleAboveName;
+            isTitleVisible = nameplateInfo.IsTitleVisible;
+            iconId = (int)nameplateInfo.IconID;
+
+            // Handle changed nodes
+            foreach (var node in nameplateInfo.Nodes)
             {
-                // Get new states
-                isTitleAboveName = nameplateInfo.IsTitleAboveName;
-                isTitleVisible = nameplateInfo.IsTitleVisible;
-                iconId = (int)nameplateInfo.IconID;
-
-                // Get new Title string content
-                var titleRaw = nameplateInfo.Title.Encode();
-                var titleNewRaw = nameplateInfo.Title.Encode();
-                if (!titleRaw.SequenceEqual(titleNewRaw))
-                    MemoryHelper.WriteSeString(titlePtr, nameplateInfo.Title);
-
-                // Get new Name string content
-                var nameRaw = nameplateInfo.Name.Encode();
-                var nameNewRaw = nameplateInfo.Name.Encode();
-                if (!nameRaw.SequenceEqual(nameNewRaw))
-                    MemoryHelper.WriteSeString(namePtr, nameplateInfo.Name);
-
-                // Get new Free Company string content
-                var freeCompanyRaw = nameplateInfo.FreeCompany.Encode();
-                var freeCompanyNewRaw = nameplateInfo.FreeCompany.Encode();
-                if (!freeCompanyRaw.SequenceEqual(freeCompanyNewRaw))
-                    MemoryHelper.WriteSeString(freeCompanyPtr, nameplateInfo.FreeCompany);
-
-                // Get new Prefix string content
-                var prefixRaw = nameplateInfo.Prefix.Encode();
-                var prefixNewRaw = nameplateInfo.Prefix.Encode();
-                if (!prefixRaw.SequenceEqual(prefixNewRaw))
-                    MemoryHelper.WriteSeString(prefixPtr, nameplateInfo.Prefix);
+                if (node.HasChanged)
+                {
+                    var textRaw = node.Text.Encode();
+                    node.Pointer = MemoryHelper.Allocate(textRaw.Length);
+                    ptrToFree.Add(node.Pointer);
```

Comment:
Btw, the reason I told you to use Utf8String is because it will add a null terminator after the last byte of the string.
The way you did it now doesn’t have that, so it doesn’t know where the string ends.
You can just push Utf8Strings to the list and then call Dispose in the ForEach

Reply:
Tried that but can't call `Dispose` as it's non-public.

Reply:
Oh, do you mean `Utfa8String.Dtor(bool)`? Missed that before.

Reply:
Might have to cast it to IDisposable, like
`(utf8string as IDisposable).Dispose()`

Reply:
Dispose calls Dtor(false), so that would work too, yep,